### DISCUSSION TITLE
'blank' contry

### DISCUSSION THREAD

**Discussion by dmatysiak94** [typesnippet/antd-phone-input#76] [Q&A]
is it possible to implement 'blank' country? I mean to enable option to display no country flag at the beggining

**Comment by ArtyomVancyan** [comment]
@dmatysiak94, do you want to make it optional or just hide the flag?

**Reply by dmatysiak94** [reply]
hey. Thanks for fast response ! I would like that after loading the page, in the flag dropdown, no flag is selected, and as a result, there is no country code in the input field. I would like it so that you have to click the dropdown list and only then select a country. Is it even possible ?

**Reply by ArtyomVancyan** [reply]
No, you cannot. Even if you specify the `excludeCountries` property of the input component, listing all the country codes to hide the initial values of the component including the flag, you won't be able to select a country because all countries have been excluded from the list.

**Reply by dmatysiak94** [reply]
are you going to add something like this in the future ? for example react-phone-number-input library has option 'International' which is not a country nor has any country code. It would Be great to have this possibility also in this library.

thanks once again for all your answers. I really appreciate it.

**Reply by ArtyomVancyan** [reply]
Maybe in the future. FWI: When you don't specify a country, it autodetects the user's country (don't try using a VPN, it works based on the user's timezone). I think it is the best option for users, the country as the one they are visiting the website from (it doesn't raise any questions), and in case they want to go with a foreign country's phone number, they can select their desired country from the dropdown (if it's not disabled) or rewrite their number and the country data will be updated.

**Reply by adrielschmitz** [reply]
Hello, to hide the country flag, I need to send a `$no_flag`, for example.

<img width="486" height="452" alt="Screenshot 2026-01-12 at 11:26:11" src="https://github.com/user-attachments/assets/620ccf0c-08ff-4ef7-b14e-56186e6b2534" />

Sometimes, I just want to show a phone number in the correct format. In that case, I don't need the country flag. What do you think about implementing this natively? Send a `noFlag` property and that's it, the flag disappears.

**Reply by ArtyomVancyan** [reply]
@adrielschmitz, if you want to hide the country flag while still beeing able to edit the phone value, you can do it by CSS manipulation (e.i. `display: none`), or if you're just interested in displaying the phone number in correct format without country flag, you can use the `getFormattedNumber` function of the https://github.com/typesnippet/react-phone-hooks.

**Reply by adrielschmitz** [reply]
Yep, that's exactly what I do (below), but it was better in native! 
```
${p => p.$no_flag && css`
    .ant-input-group-addon:has(.ant-select) {
      display: none;
    }
  `}
```
